NetProbe/Model/NetworkMeasurement.mm

```diff
@@ -152,9 +153,17 @@ - (void) run {
         .set_input_filepath([path UTF8String])
         .set_output_filepath([[self getFileName:@"json"] UTF8String])
         .set_verbosity(MK_LOG_INFO)
-        .on_log([self](uint32_t, const char *s) {
+        .on_log([self](uint32_t type, const char *s) {
             NSString *current = [NSString stringWithFormat:@"%@: %@", [super getDate], [NSString stringWithUTF8String:s]];
             NSLog(@"%s", s);
+            if ((type & MK_LOG_JSON) != 0) {
+                NSData *data = [[NSString stringWithUTF8String:s] dataUsingEncoding:NSUTF8StringEncoding];
+                NSDictionary *json = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
+                self.progress = [[json objectForKey:@"progress"] floatValue];
+                dispatch_async(dispatch_get_main_queue(), ^{
+                    [[NSNotificationCenter defaultCenter] postNotificationName:@"reloadTable" object:nil];
+                });
+            }
```

Comment:
Yes, this could happen. My suggestion is to code defensively and handle gracefully the case where the "progress" key is not available.